simpervisor/process.py

```diff
@@ -210,7 +222,9 @@ async def ready(self):
             # FIXME: We should probably check again if our process is still running
             # FIXME: Should we be locking something here?
             try:
-                is_ready = await asyncio.wait_for(self.ready_func(self), 1)
+                # Timeout of 5 secs is needed as DNS resolution of localhost
+                # on Windows takes significant time.
+                is_ready = await asyncio.wait_for(self.ready_func(self), 5)
```

Comment:
Really? :fearful: 

If it's the DNS that's slow, is it any quicker to use 127.0.0.1 instead of localhost?

Reply:
Using `127.0.0.1` is definitely faster, however that would mean we are hardcoding this to IPv4. Similar concern has been raised in https://github.com/jupyterhub/jupyter-server-proxy/pull/181.

The value of 5 seconds used here is more of a precaution for customers using low-spec hardware. On a modern Windows machine, the worst I have seen is 1-2 seonds.